### PR TITLE
feat: Add CI/CD to template

### PR DESCRIPTION
**Pull request by daniel-makerx**
Resolves 

https://github.com/algorandfoundation/algokit-cli/issues/257
https://github.com/algorandfoundation/algokit-cli/issues/258

### CODE REVIEW

File: template_content/{% if use_github_actions %}.github{% endif %}/workflows/check-python.yaml.jinja [hunk 1]
```diff
@@ -0,0 +1,99 @@
+name: Check code base
```
Comment (robdmoore) [inline template_content/{% if use_github_actions %}.github{% endif %}/workflows/check-python.yaml.jinja:1]: Minor: Name of this file is not quite right if you have typescript FYI

Comment (daniel-makerx) [reply template_content/{% if use_github_actions %}.github{% endif %}/workflows/check-python.yaml.jinja:1]: ah of course, thanks

File: template_content/README.md.jinja [hunk 2]
```diff
@@ -40,6 +40,67 @@ This project has been generated using AlgoKit. See below for default getting sta
 1. If you update to the latest source code and there are new dependencies you will need to run `algokit bootstrap all` again
 2. Follow step 3 above
 
+{% if use_github_actions -%}
+### Continuous Integration
+
+  This project uses [GitHub actions](https://docs.github.com/en/actions/learn-github-actions/understanding-github-actions) to define CI/CD workflows, see the `.github/workflows` folder for details.
```
Comment (achidlow) [inline template_content/README.md.jinja:46]: "for details" implies some sort of explanation will be found there?

File: template_content/{% if use_github_actions %}.github{% endif %}/workflows/check-python.yaml.jinja (hunk 1 above)
Comment (daniel-makerx) [reply template_content/{% if use_github_actions %}.github{% endif %}/workflows/check-python.yaml.jinja]: Made the filename and descriptions language agnostic

File: template_content/README.md.jinja (hunk 2 above)
Comment (daniel-makerx) [reply template_content/README.md.jinja]: Fixed

**Pull request merged**